pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -96,6 +96,8 @@ public RealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetadat
       RealtimeTableDataManager realtimeResourceManager, final String resourceDataDir, final ReadMode mode,
       final Schema schema, final ServerMetrics serverMetrics) throws Exception {
     super();
+    final String segmentVersionStr = tableConfig.getIndexingConfig().getSegmentFormatVersion();
```

Comment:
If you see a few lines below, I call the method SegmentVersion.fromStringOrDefault() method, which takes care to handle null or unknown segment version, and returns a (typed) segmentVersion